PhotoApp/MainViewController/MainViewController.swift

```diff
@@ -131,269 +154,246 @@ class MainViewController: UIViewController,UIGestureRecognizerDelegate,UIImagePi
-        alert.addAction(UIAlertAction.init(title: "Take A Picture", style: UIAlertAction.Style.default, handler: { (UIAlertAction) in
+
+    func addPhoto() {
+        let alert = UIAlertController.init(title: "Add photo", message: nil, preferredStyle: UIAlertController.Style.actionSheet)
+        alert.addAction(UIAlertAction.init(title: "Take A Picture", style: UIAlertAction.Style.default, handler: { (_) in
             self.openCamera()
         }))
-        alert.addAction(UIAlertAction.init(title: "Choose From Gallery", style: UIAlertAction.Style.default, handler: { (UIAlertAction) in
+        alert.addAction(UIAlertAction.init(title: "Choose From Gallery", style: UIAlertAction.Style.default, handler: { (_) in
             self.openGallery()
         }))
         alert.addAction(UIAlertAction.init(title: "Cancel", style: UIAlertAction.Style.cancel, handler: nil))
-       
+
         self.present(alert, animated: true, completion: nil)
-        
+
     }
-    
-    func openCamera(){
-       // UIWindow.main.view
-        
-        if(UIImagePickerController.isSourceTypeAvailable(UIImagePickerController.SourceType.camera)){
+
+    func openCamera() {
+        if UIImagePickerController.isSourceTypeAvailable(UIImagePickerController.SourceType.camera) {
             let status = AVCaptureDevice.authorizationStatus(for: AVMediaType.video)
-            if (status == .denied){
-                
-            }
-            else{
+            if status == .denied {} else {
                 self.imagePickerController.sourceType = UIImagePickerController.SourceType.camera
                 self.imagePickerController.allowsEditing = false
-                if (self.imagePickerController != nil){
                 self.present(self.imagePickerController, animated: true, completion: nil)
-                }
+
             }
         }
     }
-    
-    
-    func openGallery(){
+
+    func openGallery() {
         let status = PHPhotoLibrary.authorizationStatus()
-        if (status == .denied){
-            
-        }
-        else{
+        if status == .denied {} else {
             self.imagePickerController.sourceType = UIImagePickerController.SourceType.photoLibrary
             self.imagePickerController.allowsEditing = false
-            if (self.imagePickerController != nil){
                 self.present(self.imagePickerController, animated: true, completion: nil)
-            }
         }
     }
-    
+
     func imagePickerControllerDidCancel(_ picker: UIImagePickerController) {
         self.dismiss(animated: true, completion: nil)
     }
-    
-    func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [UIImagePickerController.InfoKey : Any]) {
-        self.dismiss(animated: true, completion: nil)
+
+    func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [UIImagePickerController.InfoKey: Any]) {
+
+        var currentdate: Date!
```

Comment:
@Kasiama could you explain, please var and "!"?

Reply:
I just tested something. Fixed